dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerImportService.java

```diff
@@ -102,32 +104,62 @@ public TrackerImportReport importTracker( TrackerImportParams params )
         {
             notifier.notify( params.getJobConfiguration(), "(" + params.getUsername() + ") Import:Start" );
         }
+        
+        try
+        {
 
-        List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
+            List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
 
-        trackerBundles = preProcessBundle( trackerBundles, importReport );
+            trackerBundles = preProcessBundle( trackerBundles, importReport );
 
-        TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
+            TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
 
-        if ( validationReport.hasErrors() && params.getAtomicMode() == AtomicMode.ALL )
-        {
-            importReport.setStatus( TrackerStatus.ERROR );
-        }
-        else
-        {
-            if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
+            if ( validationReport.hasErrors() && params.getAtomicMode() == AtomicMode.ALL )
             {
-                deleteBundle( params, importReport, trackerBundles );
+                importReport.setStatus( TrackerStatus.ERROR );
             }
             else
             {
-                commitBundle( params, importReport, trackerBundles );
+                if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
+                {
+                    deleteBundle( params, importReport, trackerBundles );
+                }
+                else
+                {
+                    commitBundle( params, importReport, trackerBundles );
+                }
             }
+
+            importReport.getTimings().setTotalImport( requestTimer.toString() );
+
+            TrackerBundleReportModeUtils.filter( importReport, params.getReportMode() );
+
         }
+        catch ( Exception e )
+        {
+            
+            TrackerValidationReport tvr = importReport.getTrackerValidationReport();
+            
+            if ( tvr == null )
+            {
+                tvr = new TrackerValidationReport();
+            }
+            
+            tvr.getErrorReports().add(new TrackerErrorReport( null, "Exception:"+e.getMessage(), TrackerErrorCode.E9999, 0, null, null, null, null ));
+            importReport.setTrackerValidationReport( tvr );
+            importReport.setStatus( TrackerStatus.ERROR );
+            
+            if ( params.hasJobConfiguration() )
+            {
+                notifier.update( params.getJobConfiguration(), "(" + params.getUsername() + ") Import:Failed with exception: " + e.getMessage(), true );
+                notifier.addJobSummary( params.getJobConfiguration(), importReport, TrackerImportReport.class );
+            }
+            
+            //TODO: Should this be rethrown in case of sync? Or is the error report added to the import report in the above lines enough?
+            throw e;
```

Comment:
I think we shouldnt rethrow this exception. Your thoughts?

Reply:
I don't think we should rethrow. If we rethrow, then the exception will be "handled" by the `SecurityContextRunnable` class (or a subclass). Please, see my comment in the approve box, based on our conversation.